### PR TITLE
Added tests for draw

### PR DESCRIPTION
**Pull request by georgeschena**
**Added unit and acceptance tests for a draw condition.**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ source "https://rubygems.org"
 
 git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
 
-# gem "rails"
+gem "rails"
```
Comment (rebeccafitzsimmons1) [inline Gemfile]: test comment

Comment (naliwajek) [inline Gemfile]: Why add `rails` ?

Comment (georgeschena) [reply Gemfile]: Thanks

File: spec/unit/outcomes_spec.rb [hunk 2]
```diff
@@ -120,4 +120,20 @@
     outcomes = Outcomes.new(mark_gateway: mark_gateway)
     expect(outcomes.execute).to eq('AI has won diagonally')
   end
+
+  it 'determines draw' do
```
Comment (georgeschena) [inline spec/unit/outcomes_spec.rb]: Test looks good

File: Gemfile (hunk 1 above)
Comment (georgeschena) [reply Gemfile]: Because were about to start implementing a UI today

Comment (naliwajek) [reply Gemfile]: Then I'd not include this now.

It's not part of the feature this PR is about.

Add this in the next PR which will be about implementing UI.

Comment (craigjbass) [reply Gemfile]: You should probably only commit this change once you have UI changes

**Comment by zuren** [comment]
Looks really good. Would recommend not including the `Gemfile` and `Gemfile.lock` changes in this PR though, as they're not really relevant to the change. I'd have saved adding Rails for when this PR was merged and you were ready to start the UI.

File: spec/acceptance/tic-tac-toe_spec.rb [hunk 3]
```diff
@@ -109,4 +109,17 @@ def mark_properties=(mark_properties)
...
+    set_a_mark.execute(type_of_mark: 'X', position_on_board: 1)
+    set_a_mark.execute(type_of_mark: 'O', position_on_board: 2)
+    set_a_mark.execute(type_of_mark: 'X', position_on_board: 3)
+    set_a_mark.execute(type_of_mark: 'X', position_on_board: 4)
+    set_a_mark.execute(type_of_mark: 'O', position_on_board: 5)
+    set_a_mark.execute(type_of_mark: 'O', position_on_board: 6)
+    set_a_mark.execute(type_of_mark: 'O', position_on_board: 7)
+    set_a_mark.execute(type_of_mark: 'X', position_on_board: 8)
+    set_a_mark.execute(type_of_mark: 'X', position_on_board: 9)
+    expect(outcomes.execute).to eq('Draw')
```
Comment (craigjbass) [inline spec/acceptance/tic-tac-toe_spec.rb]: I'd expect `outcomes.execute` to return a hash

So this line would become `expect(outcomes.execute).to eq({status: 'Draw'})`

**Pull request merged**